packages/commerce-sdk-react/src/hooks/createUseQuery.ts

```diff
@@ -0,0 +1,154 @@
+    queryOptions?: ApiQueryOptions<M>
+) => UseQueryResult<DataType<M>, Error>
+
+/**
+ * Helper to ensure a method is an ApiMethod
+ */
+type EnsureApiMethod<T> = T extends ApiMethod<any, any> ? T : never
+
+/**
+ * Creates a typed query hook for a specific API client method.
+ *
+ * @template ClientKey - The key of the client in ApiClients
+ * @template M - The method name from the specified client
+ * @param options - Configuration options for creating the query hook
+ * @returns A custom hook that provides access to the specified API method
+ */
+export const createUseQuery = <
+    ClientKey extends keyof ApiClients,
+    M extends MethodsOf<ApiClients[ClientKey]>
+>(
+    options: CreateUseQueryOptions<ClientKey, M>
+): QueryHook<EnsureApiMethod<ApiClients[ClientKey][M]>> => {
+    const {clientKey, methodName, displayName, queryKeyHelper} = options
+    type Client = ApiClients[ClientKey]
+    // Ensure method type is a function
+    type Method = ApiClients[ClientKey][M] & ApiMethod<any, any>
+
+    /**
+     * Custom hook for accessing an API method
+     */
+    const useQueryHook = (
+        apiOptions: NullableParameters<Argument<Method>>,
+        queryOptions: ApiQueryOptions<Method> = {}
+    ): UseQueryResult<DataType<Method>, Error> => {
+        type Options = Argument<Method>
+        type Data = DataType<Method>
+
+        // Get the client from useCommerceApi based on clientKey
+        const commerceApi = useCommerceApi()
+        const client: Client = commerceApi[clientKey]
+
+        // Get the SDK class for parameter keys
+        const SDKClass = client.constructor as unknown as SDKClassConstructor
+
+        // Get required parameters for this method
+        const requiredKey = `${String(methodName)}Required`
+        const requiredParameters = SDKClass.paramKeys[requiredKey]
+
+        const netOptions = omitNullableParameters(mergeOptions(client, apiOptions))
+
+        // Get valid parameter keys for this method
+        const methodParamKeys = SDKClass.paramKeys[String(methodName)]
+
+        // These parameters are valid at runtime
+        const parameters = pickValidParams(netOptions.parameters || {}, methodParamKeys)
+
+        // Get query key for this method
+        const queryKey = queryKeyHelper.queryKey(netOptions.parameters)
+
+        // We don't use `netOptions` here because we manipulate the options in `useQuery`.
+        const method = async (options: Options) => {
+            // client[methodName] is guaranteed to be a function at runtime
+            return await (client[methodName] as Method)(options)
+        }
+
+        queryOptions.meta = {
+            displayName,
+            ...queryOptions.meta
+        }
+
+        return useQuery<
+            typeof client,
+            Options,
+            Data,
+            Error,
+            Data // We're not transforming the data, so TData = TQueryFnData
+        >({...netOptions, parameters} as any, queryOptions as any, {
```

Comment:
🤔 It's interesting that we have `as any` here on this line. 

For example: `queryOptions` in the following places are typed differently. Since we have control over them, would it be better if they use the same type? If yes, then we can probably remove the casting to `any`.

- https://github.com/SalesforceCommerceCloud/pwa-kit/blob/da431ac18941673307d565f05cd4d09d4cdb25fb/packages/commerce-sdk-react/src/hooks/createUseQuery.ts#L102
- https://github.com/SalesforceCommerceCloud/pwa-kit/blob/da431ac18941673307d565f05cd4d09d4cdb25fb/packages/commerce-sdk-react/src/hooks/useQuery.ts#L49-L52

Reply:
@kevinxh It turns out that we can remove casting of `queryOptions` by making the **error type** consistent. Take a look at #2368 where the the generic `TError` is defined and propagated throughout.